File: 27_spring_cache/local_cache/src/main/java/com/yologger/local_cache/repository/MemberEntity.java
package com.yologger.local_cache.repository;

import com.yologger.local_cache.service.MemberDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name= "member")
@Getter
@NoArgsConstructor
public class MemberEntity {

    @Id
    @Column(name="id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String email;

    @Column
    private String password;

    @Builder
    public MemberEntity(Long id, String email, String password) {
        this.id = id;
        this.email = email;
        this.password = password;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public MemberDTO toDTO() {
        return MemberDTO.builder()
                .id(this.id)
                .email(this.email)
                .password(this.password)
                .build();
    }
}

File: 14_spring_webflux/02_webflux_kotlin/webflux_kotlin/src/main/kotlin/com/yologger/samples/webflux_kotlin/repository/MemberEntity.kt
package com.yologger.samples.webflux_kotlin.repository

import com.yologger.samples.webflux_kotlin.service.MemberDTO
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table

@Table("member")
data class MemberEntity (
    @Id var id: Long? = null,
    @Column var email: String,
    @Column var name: String,
    @Column val password: String
) {
    fun toDTO(): MemberDTO {
        return MemberDTO(id = id, email = email, name = name, password = password)
    }
}